Python/Product/Analysis/Interpreter/Ast/NameLookupContext.cs

```diff
@@ -464,8 +462,14 @@ class NameLookupContext {
             if (expr is DictionaryExpression || expr is DictionaryComprehension) {
                 return Interpreter.GetBuiltinType(BuiltinTypeId.Dict);
             }
-            if (expr is TupleExpression) {
-                return Interpreter.GetBuiltinType(BuiltinTypeId.Tuple);
+            if (expr is TupleExpression tex) {
+                var types = tex.Items
+                    .Where(t => t is NameExpression)
```

Comment:
It seems like this is going to create a tuple of length n-m when m components are not names, and not in the array.  Instead of .Where's here it seems like we want only do a select, and avoid the where's, and produce an empty result of some form for the types we can't statically resolve

Reply:
OK